packages/vow/test/test-watch-utils.js

```diff
@@ -0,0 +1,91 @@
+// @ts-check
+import test from 'ava';
+
+import { makeHeapZone } from '@agoric/base-zone/heap.js';
+
+import { prepareVowTools } from '../src/tools.js';
+
+test('allVows waits for a single vow to complete', async t => {
+  const zone = makeHeapZone();
+  const { watch, when, allVows } = prepareVowTools(zone);
+
+  const testPromiseP = Promise.resolve('promise');
+  const vowA = when(watch(testPromiseP));
```

Comment:
For all of these input vows (like `vowA`, `vowB`, `vowC`), you should not need to use `when(watch(...))` as that returns a promise for the final result, never an intermediate vow.  Can you please change the input vows to be just `watch(testPromiseAP)` etc?

Do note that a `watch promises directly` test, such as `when(allVows(Promise.resolve('first'), Promise.resolve('second')))` is still valuable as a separate test, as well as `watch promises mixed with vows` is too.